### PR TITLE
Improve performance and handle over 100 saml identities

### PR DESCRIPTION
**Pull request by pholleran**
## TL; DR

This PR improves the performance of the queries sent to the GraphQL API and handles more than 100 SAML identities in an organization

### Background

The existing code uses a single [organizationQuery](https://github.com/SvanBoxel/org-audit-action/blob/af1cee26853864ab4f4e8d0deb4b3de1afaf41e7/src/queries.js#L4) to retrieve SAML identities, repositories, and collaborators. As a result duplicate SAML identities are fetched with each request, which slows the query performance.

Additionally, the existing query [hardcodes](https://github.com/SvanBoxel/org-audit-action/blob/af1cee26853864ab4f4e8d0deb4b3de1afaf41e7/src/queries.js#L8) 100 `externalIdentities` and doesn't use pagination to retrieve more if they exist.

### Solution

* The `organizationQuery` is renamed to `orgRepoAndCollaboratorQuery` and all SAML identity components are removed
* A new `orgSAMLquery` is added, with pagination support, to retrieve SAML identities
* The `requestOrganizationData` function is renamed to `requestOrgReposAndCollaborators` to reflect the new function's focus on retrieving only repos and their collaborators
* New `requestSAMLidentities` and `collectSAMLidentities` functions are added, following the existing request/collect pattern for dealing with GraphQL results
* the new `collectSAMLidentities` function is called within the for each org loop in `startCollection()` after `collectData()` has completed

### Data

* this [workflow run](https://github.com/octodemo/example-workflows/actions/runs/1141728891) of the existing code analyzed the `octodemo` org in 1 hour 8 minutes
* this [workflow run](https://github.com/octodemo/example-workflows/actions/runs/1156344815) of my branch analyzed the same octodemo org in 43 minutes 55 seconds

> note: both workflow runs failed to upload the artifacts, suffering connection resets. This did not happen with data from a much smaller (fewer members and repos) org

I have also [verified](https://github.com/se-saml/example-workflows/runs/3394238212?check_suite_focus=true) pagination of SAML identities works correctly by testing locally against a lower `first:` number.

### Additional Investigation

* If this PR is accepted I may try to get some further performance improvements by exploring the use of the REST API to fetch all repositories and collaborators. `octokit/rest.js` has built in pagination, which keeps things simple, and leverages a single endpoint. This *may* lead to further performance improvements since GraphQL isn't needed to stitch together disparate data elements.

* Uploading the .csv contents to the issue fails when the .csv is too large. This makes sense as the issue shouldn't be expected to contain tens of thousands of rows. I might consider taking the approach in https://github.com/gr2m/github-organization-repository-auditing-action#usage and committing the file to the repo, creating a git log of all access. Alternatively, uploading the artifact as a file to an issue, instead of rendered markdown, may be less error prone than uploading to the workflow run.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -271,18 +336,16 @@ class CollectUserData {
...
-    );
+    await writeFileAsync(`${DATA_FOLDER}/${ARTIFACT_FILE_NAME}.csv`, csv);
 
     await this.createandUploadArtifacts();
     await this.postResultsToIssue(csv);
     process.exit();
   }
 
-  normalizeResult() {
+  async normalizeResult() {
```
Comment (SvanBoxel) [inline src/index.js:346]: Any reason you made this an `async` method? I can't spot any asynchronous in the method's code.

**Review by SvanBoxel — APPROVED** [review]
Looks great @pholleran ! 🙇🏽

File: src/index.js (hunk 1 above)
Comment (pholleran) [reply src/index.js:346]: Thanks for catching that. I made it async when I put the code to handle the SAML identities inside `normalizeResult()`. I ended up moving it out, but forgot to remove it.
